yargs.js

```diff
@@ -172,56 +172,129 @@ function Yargs (processArgs, cwd, parentRequire) {
-    if (typeof key === 'object') {
-      Object.keys(key).forEach(function (k) {
-        self.choices(k, key[k])
-      })
-    } else {
-      options.choices[key] = (options.choices[key] || []).concat(values)
+  self.count = function (keys) {
+    populateParserHintArray('count', keys)
+    return self
+  }
+
+  self.string = function (keys) {
+    populateParserHintArray('string', keys)
+    return self
+  }
+
+  self.requiresArg = function (keys) {
+    populateParserHintArray('requiresArg', keys)
+    return self
+  }
+
+  self.skipValidation = function (keys) {
+    populateParserHintArray('skipValidation', keys)
+    return self
+  }
+
+  function populateParserHintArray (type, keys, value) {
+    keys = [].concat(keys)
+    keys.forEach(function (key) {
+      self.global(key)
+      options[type].push(key)
+    })
+  }
+
+  self.nargs = function (key, value) {
+    populateParserHintObject(self.nargs, false, 'narg', key, value)
+    return self
+  }
+
+  self.choices = function (key, value) {
+    populateParserHintObject(self.choices, true, 'choices', key, value)
+    return self
+  }
+
+  self.alias = function (key, value) {
+    populateParserHintObject(self.alias, true, 'alias', key, value)
+    return self
+  }
+
+  // TODO: actually deprecate self.defaults.
+  self.default = self.defaults = function (key, value, defaultDescription) {
+    if (defaultDescription) options.defaultDescription[key] = defaultDescription
+    if (typeof value === 'function') {
+      if (!options.defaultDescription[key]) options.defaultDescription[key] = usage.functionDescription(value)
+      value = value.call()
     }
+    populateParserHintObject(self.default, false, 'default', key, value)
     return self
   }
 
-  self.normalize = function (strings) {
-    options.normalize.push.apply(options.normalize, [].concat(strings))
+  self.describe = function (key, desc) {
+    populateParserHintObject(self.describe, false, 'key', key, true)
+    usage.describe(key, desc)
     return self
   }
 
+  self.demandOption = function (keys, msg) {
+    var value = { msg: typeof msg === 'string' ? msg : undefined }
+    populateParserHintObject(self.demandOption, false, 'demandedOptions', keys, value)
+    return self
+  }
+
+  self.coerce = function (keys, value) {
+    populateParserHintObject(self.coerce, false, 'coerce', keys, value)
+    return self
+  }
+
+  function populateParserHintObject (builder, isArray, type, key, value) {
+    if (Array.isArray(key)) {
+      // an array of keys with one value ['x', 'y', 'z'], function parse () {}
+      var temp = {}
+      key.forEach(function (k) {
+        temp[k] = value
+      })
+      builder(temp)
+    } else if (typeof key === 'object') {
+      // an object of key value pairs: {'x': parse () {}, 'y': parse() {}}
+      Object.keys(key).forEach(function (k) {
+        builder(k, key[k])
+      })
+    } else {
+      // a single key value pair 'x', parse() {}
+      self.global(key)
```

Comment:
Similarly, this approach means that you _must_ call `.global(opts, false)` _last_ to disable an option from being global.

E.g. code like this will not have the desired effect because the call to `.coerce()` makes the option global again:

```js
require('yargs')
  .describe('opt', 'My local option')
  .string('one')
  .global('opt', false)
  .coerce('opt', opt => 'local_' + opt)
  .argv
```

This is probably not a big deal, but we should either mention this caveat in the README/docs or go with another approach that doesn't have this caveat.